tools/db-predeclare.py

```diff
@@ -0,0 +1,268 @@
+    ) -> List[Metric]:
+        vhost = self.data_vhost_quoted()
+        async with http_session.get(
+            self.http_api_url.with_path(
+                f"/api/bindings/{vhost}/e/{exchange}/q/{queue}/", encoded=True
+            )
+        ) as response:
+            return [binding["routing_key"] for binding in await response.json()]
+
+    async def fetch_bindings(self, db_token: str) -> Tuple[List[Metric], List[Metric]]:
+        auth = (
+            aiohttp.BasicAuth(
+                login=self.http_api_url.user, password=self.http_api_url.password
+            )
+            if self.http_api_url.user and self.http_api_url.password
+            else None
+        )
+        data_queue = f"{db_token}-data"
+        hreq_queue = f"{db_token}-hreq"
+
+        async with ClientSession(auth=auth) as session:
+            data_bindings, history_bindings = await asyncio.gather(
+                self.fetch_queue_bindings(
+                    http_session=session,
+                    exchange=self.data_exchange,
+                    queue=data_queue,
+                ),
+                self.fetch_queue_bindings(
+                    http_session=session,
+                    exchange=self.history_exchange,
+                    queue=hreq_queue,
+                ),
+            )
+
+            return data_bindings, history_bindings
+
+    async def preregister(self, db_token: str):
+        data_bindings, history_bindings = await self.fetch_bindings(db_token=db_token)
+        logger.info(
+            "Database {!r} had {} data and {} history binding(s)",
+            db_token,
+            len(data_bindings),
+            len(history_bindings),
+        )
+
+        if self.dry_run:
+            logger.info(
+                "Dry-run: would have declared the following for database {!r}:",
+                db_token,
+            )
+            logger.info("Data bindings: {}", data_bindings)
+            logger.info("History bindings: {}", history_bindings)
+        else:
+            logger.info("Pre-declaring queues for database {!r}", db_token)
+            await self.queue_manager.db_declare_queues(
+                db_token=db_token,
+                data_bindings=data_bindings,
+                history_bindings=history_bindings,
+            )
+
+    async def stop(self, exception):
+        if self.couchdb_client:
+            await self.couchdb_client.close()
+        if self.queue_manager:
+            await self.queue_manager.close()
+        await super().stop(exception)
+
+
+@click.command()
+@click_log.simple_verbosity_option(logger, default="warning")
+@click.option(
+    "--server",
+    default="amqp://admin:admin@localhost/",
+    help="Management address of the MetricQ server",
+)
+@click.option(
+    "--rabbitmq-api",
+    default=None,
+    help="RabbitMQ API address of the MetricQ server, will be derived from --server if omitted",
+)
+@click.option(
+    "--data-vhost",
+    default="/",
```

Comment:
Isn't that `data`?

Reply:
Well, probably. But the default `Dockerfile`s do not have vhosts set up and the default value for `--data-url` of the manager uses the root vhost.

Reply:
Consensus seemed to be that in a live situation, giving the argument is a good idea.